flyteadmin/auth/authzserver/metadata_provider.go

```diff
@@ -107,3 +121,45 @@ func NewService(config *authConfig.Config) OAuth2MetadataProvider {
 		cfg: config,
 	}
 }
+
+func sendAndRetryHttpRequest(ctx context.Context, client *http.Client, url string, retryAttempts int, retryDelay time.Duration) (*http.Response, error) {
+	var response *http.Response
+	var err error
+	totalAttempts := retryAttempts + 1 // Add 1 for the case where 0 retryAttempts are specified
+
+	backoff := wait.Backoff{
+		Duration: retryDelay,
+		Steps:    totalAttempts,
+		Cap:      0,
```

Comment:
is it necessary to set this to the 0 value?

Reply:
saw this too! removed!